account_payment_batch_process_discount/models/account_payment_term.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("discount_days"):
+            for term in self:
+                # get all invoice related to this payment term and update
+                # validity discount date
+                invoice_ids = self.env["account.move"].search(
+                    [
+                        ("state", "=", "posted"),
+                        ("invoice_payment_term_id", "=", term.id),
+                    ]
+                )
+                for inv in invoice_ids:
+                    # Check payment date discount validation
+                    invoice_date = fields.Date.from_string(inv.invoice_date)
```

Comment:
```suggestion
                    invoice_date = inv.invoice_date
```
inv.invoice_date is a datetime.date instance already.